### PR TITLE
Empty dicts to use current active plotstyle

### LINKED ISSUE #400: Context manager for plots is not working with dicts
## General

* pyfar version: 0.5.1

## Description

It seems like  the plot functions work in combination with_pyfar.plot.context()_ (and _matplotlib.style.context()_) with styles as input:

```
import pyfar as pf
import matplotlib as mpl
import matplotlib.pyplot as plt
pf.plot.use()

with pf.plot.context('classic'):
    fig, ax = plt.subplots(2, 1)
    pf.plot.time(pf.TimeData([0, 1, 0, -1], range(4)), ax=ax[0])
    ax[1].plot(range(4), [0, 1, 0, -1])
```
grafik

But not with dicts as arguments:
```
import pyfar as pf
import matplotlib as mpl
import matplotlib.pyplot as plt
pf.plot.use()

with pf.plot.context({'lines.linestyle': ':'}):
    fig, ax = plt.subplots(2, 1)
    pf.plot.time(pf.TimeData([0, 1, 0, -1], range(4)), ax=ax[0])
    ax[1].plot(range(4), [0, 1, 0, -1])
```
grafik

I suppose the reason is that for example the linestyle is overwritten by the call of _pyfar.plot.context()_ within the plot function. In constrast, this works well:
```
import pyfar as pf
import matplotlib as mpl
import matplotlib.pyplot as plt
pf.plot.use()

style = {'lines.linestyle': ':'}

fig, ax = plt.subplots(2, 1)
pf.plot.time(pf.TimeData([0, 1, 0, -1], range(4)), ax=ax[0], style=style)
ax[1].plot(range(4), [0, 1, 0, -1])
```
grafik

### PR DESCRIPTION
**Pull request by twennemann**
### Which issue(s) are closed by this pull request?

Closes #400

### Changes proposed in this pull request:

- Expanded docstring to  pass an empty dictonary ``style = {}`` to use the current active plotstyle.
- Implemented Test

### CODE REVIEW

File: tests/test_plot.py [hunk 1]
```diff
@@ -562,3 +562,15 @@ def test_title_style(style, handsome_signal):
...
                      file_type, compare_output)
     plt.close('all')
+
+
+@pytest.mark.parametrize('rcParams, value', [
+    ['lines.linestyle', ':'],
+    ['axes.facecolor', 'black'],
+    ['axes.grid', False]])
+def test_context_manager_empty_dicts(rcParams, value):
+    # Test calling empty dict to use current active plotstyle
```
Comment (f-brinkmann) [inline tests/test_plot.py:572]:
```suggestion
def test_pyfar_plot_with_empty_style(rcParams, value):
    # Test passing an empty style to a pyfar plot function to check if the currently active plot stlye remains active
```

**Review by f-brinkmann — APPROVED** [review]
Thanks for implementing. I would only suggest a more clear wording in the test.

**Review by sikersten — APPROVED** [review]

**Review by f-brinkmann — DISMISSED** [review]

File: pyfar/plot/line.py [hunk 2]
```diff
@@ -47,7 +47,10 @@ def time(signal, dB=False, log_prefix=20, log_reference=1, unit="s",
         or creates a new figure if none exists.
     style : str
         ``light`` or ``dark`` to use the pyfar plot styles or a plot style from
-        ``matplotlib.style.available``. The default is ``light``.
+        ``matplotlib.style.available``. Pass a dictonary to set specific plot
+        parameters, for example ``style = {'axes.facecolor':'black'}``. Pass an
+        empty dictonary ``style = {}`` to use the current active plotstyle.
```
Comment (f-brinkmann) [inline pyfar/plot/line.py:52]: Just a tiny correction, I think it should be
```suggestion
        empty dictonary ``style = {}`` to use the currently active plotstyle.
```
sorry for not noting this earlier

**Review by f-brinkmann — DISMISSED** [review]
Found only one typo. Approved otherwise. Thanks for the effort

**Review by f-brinkmann — APPROVED** [review]

**Review by mberz — COMMENTED** [review]
Looks good.
One additional idea, which is optional: Could you maybe add a bit of documentation on setting the style in the concepts section of the docs?

File: pyfar/_concepts/plots.py [hunk 3]
```diff
@@ -19,6 +19,40 @@
...
+to overwrite the currently used plot style or
+
+::
+
+    with pyfar.plot.context():
+        # everything inside the with statement
+        # uses the pyfar plot style
+
+If you want do not want to use the pyfar plot style, you can pass an empty
+dictionary to the plot functions
```
Comment (twennemann) [inline pyfar/_concepts/plots.py:43]: If you do not want*

Comment (f-brinkmann) [reply pyfar/_concepts/plots.py:43]: ups - thanks

**Review by f-brinkmann — APPROVED** [review]

**Pull request merged**